tasks/message/message.go

```diff
@@ -278,13 +279,31 @@ func (p *MessageProcessor) extractMessageModels(ctx context.Context, ts *types.T
 			}
 			result.Messages = append(result.Messages, msg)
 
+			msgsSeen[message.Cid()] = struct{}{}
+
 			if p.parseMessages {
 				dstAddr, err := address.NewFromString(msg.To)
 				if err != nil {
 					return nil, nil, xerrors.Errorf("parse to address: %w", err)
 				}
 
-				st, err := state.LoadStateTree(p.node.Store(), ts.ParentState())
+				child, err := p.node.ChainGetTipSetByHeight(ctx, ts.Height()+1, types.NewTipSetKey())
+				if err != nil {
+					// If we aren't finalized, we fail for now, because a child tipset may occur
+					if head, err := p.node.ChainGetTipSet(ctx, types.NewTipSetKey()); err == nil && head.Height()-ts.Height() < 900 {
+						log.Warn("Failing derivation for messate that does not yet have children, but might get them later")
```

Comment:
```suggestion
						log.Warn("Delaying derivation for message which is not yet finalized")
```
Is this error misleading? Looks like it's not because of lack of children, but more that it's still suseptible to reorg.